cmd/vic-machine/create/create.go

```diff
@@ -1088,26 +1080,15 @@ func (c *Create) loadCertificates() ([]byte, *certificate.KeyPair, error) {
 			log.Warnf("Unable to load client certificate - validation of API endpoint will be best effort only: %s", err)
 		}
 
-		clientCert, err := cpair.Certificate()
-		if err != nil || clientCert.Leaf == nil {
-			log.Debugf("Unable to parse client certificate: %s", err)
-			return certs, keypair, nil
-		}
-
-		pool := x509.NewCertPool()
-		if !pool.AppendCertsFromPEM(certs) {
-			log.Debugf("Unable to create CA pool to check client certificate")
-			return certs, keypair, nil
-		}
-
-		opts := x509.VerifyOptions{
-			Roots:     pool,
-			KeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
-		}
-
-		_, err = clientCert.Leaf.Verify(opts)
+		clientCert, err := certificate.VerifyClientCert(certs, cpair)
 		if err != nil {
-			log.Warnf("Client certificate in certificate path does not validate with provided CA - continuing without client certificate")
+			switch err.(type) {
+			case certificate.CertParseError, certificate.CreateCAPoolError:
+				log.Debugf(err.Error())
```

Comment:
What happens here from the user's perspective? Is this an error that should be suppressed? I assume in this case we generate certs or something so there's a reason we're not displaying this error to the user outside of debug mode?